### DISCUSSION TITLE
Authorization for self-host solutions

### DISCUSSION THREAD

**Discussion by Samsonov0** [supabase/supabase#13151] [Questions]
How can I implement the authorization and registration logic used by the cloud version of Supabase in a self-hosted environment in order to restrict unauthorized access without relying on a VPN or proxy?

**Comment by Samsonov0** [comment]
Is it possible to do this, or does supabase selfhost not have such functionality?

**Reply by GaryAustin1** [reply]
Self hosting is bring your own protection for studio.

**Reply by Samsonov0** [reply]
Do I understand correctly that you mean that the protection falls on our responsibility?

**Reply by GaryAustin1** [reply]
For self hosted projects you have to protect studio.  That is my understanding at least (I don't do self-hosted).
I assume you are talking access to the dashboard/studio.

**Reply by GaryAustin1** [reply]
If you are talking about using Auth features for users to log in and access the database (like thru client APIs) then that is supported.

**Reply by Samsonov0** [reply]
I'm talking about studio. Thanks

**Reply by GaryAustin1** [reply]
Studio is now going behind Kong for protection...
https://github.com/supabase/supabase/pull/16097
